x-pack/plugins/session_view/public/components/process_tree_node/index.tsx

```diff
@@ -126,6 +131,22 @@ export function ProcessTreeNode({
     shouldAddListener: hasInvestigatedAlert,
   });
 
+  const alertTypeCounts = useMemo(() => {
+    const alertCounts: AlertTypeCount[] = [];
+    alerts?.forEach((alert: ProcessEvent, i) => {
```

Comment:
Nitpick, as this works fine, but might be cleaner to output a map using Array.reduce?
e.g

const alertTypeCounts = alerts?.reduce((previous, current) => {
  const category = current.event?.category?.[0] as ProcessEventAlertCategory;
  previous[category]++;
  return previous;
}, { process: 0, file: 0, network: 0});

or something to that effect. You could output an array with object { category, count } but would be a few more steps to update the counts on each interation.

Reply:
or, maybe a lodash.groupBy function would work too.

e.g   groupBy(alerts, (alert) => alert.event?.category?.[0]);

Reply:
Make sense, at first tried the  `array.reduce` was giving typescript issues with my parameters,  then I went with this approach which isn't the cleanest. I agree. I'll check out lodash `groupBy`. Thanks!